qdb/tracer.py

```diff
@@ -450,6 +503,109 @@ def set_quit(self):
         """
         self.quitting = True
 
+    def eval_(self, code, pprint=False):
+        repr_fn = self.repr_fn
+
+        outexc = None
+        outmsg = None
+        with capture_output() as (out, err), \
```

Comment:
eh, I find that `ExitStack` opens up the reader to the possibility  of a dynamic stack and makes the intent less clear. We have a static 3 contexts with no conditional additions.